cudax/include/cuda/experimental/__memory_resource/legacy_pinned_memory_resource.cuh

```diff
@@ -0,0 +1,120 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDA__MEMORY_RESOURCE_LEGACY_PINNED_MEMORY_RESOURCE_H
+#define _CUDA__MEMORY_RESOURCE_LEGACY_PINNED_MEMORY_RESOURCE_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#if _CCCL_CUDA_COMPILER(CLANG)
+#  include <cuda_runtime.h>
+#  include <cuda_runtime_api.h>
+#endif // _CCCL_CUDA_COMPILER(CLANG)
+
+#include <cuda/__memory_resource/properties.h>
+#include <cuda/std/__concepts/concept_macros.h>
+#include <cuda/std/__cuda/api_wrapper.h>
+#include <cuda/std/detail/libcxx/include/stdexcept>
+
+#include <cuda/experimental/__memory_resource/memory_resource_base.cuh>
+#include <cuda/experimental/__memory_resource/pinned_memory_pool.cuh>
+
+// Trigger a rebuild of the file
+
+//! @file
+//! The \c legacy_pinned_memory_resource class provides a memory resource that allocates pinned memory.
+namespace cuda::experimental
+{
+
+//! @brief legacy_pinned_memory_resource uses `cudaMallocHost` / `cudaFreeHost` for allocation / deallocation.
```

Comment:
We should add a sentence that this is deprecated for CTKs that support asynchronous pinned memory and explain what a user should use